components/layout_2020/flexbox/layout.rs

```diff
@@ -444,7 +445,81 @@ impl<'a> FlexItem<'a> {
         let max_size = box_style.content_max_box_size(containing_block, &pbm);
         let min_size = box_style.content_min_box_size(containing_block, &pbm);
 
-        let min_size = min_size.auto_is(|| automatic_min_size(box_));
+        // https://drafts.csswg.org/css-flexbox/#min-size-auto
+        let automatic_min_size = || {
+            if cross_axis_is_item_block_axis {
+                // FIXME(stshine): Consider more situations when auto min size is not needed.
+                if box_style.get_box().overflow_x.is_scrollable() {
+                    return Length::zero();
+                }
+
+                let specified_size_suggestion = content_box_size.inline;
+
+                let transferred_size_suggestion = match box_ {
+                    IndependentFormattingContext::NonReplaced(_) => None,
+                    IndependentFormattingContext::Replaced(ref bfc) => {
+                        match (
+                            bfc.contents
+                                .inline_size_over_block_size_intrinsic_ratio(box_style),
+                            content_box_size.block,
+                        ) {
+                            (Some(ratio), LengthOrAuto::LengthPercentage(block_size)) => {
+                                let block_size = block_size.clamp_between_extremums(
+                                    min_size.block.auto_is(|| Length::zero()),
+                                    max_size.block,
+                                );
+                                Some(block_size * ratio)
+                            },
+                            _ => None,
+                        }
+                    },
+                };
+
+                let inline_content_size = box_
+                    .inline_content_sizes(&flex_context.layout_context)
+                    .min_content;
+                let content_size_suggestion = match box_ {
+                    IndependentFormattingContext::NonReplaced(_) => inline_content_size,
+                    IndependentFormattingContext::Replaced(ref replaced) => {
+                        if let Some(ratio) = replaced
+                            .contents
+                            .inline_size_over_block_size_intrinsic_ratio(box_style)
+                        {
+                            let (min_inline_size, max_inline_size) = transferred_inline_min_max(
```

Comment:
This uses the computed min/max block sizes, so it needs to check `box-sizing`. Why not simply use `min_size.block` and `max_size.block` like for the transferred size suggestion?

Reply:
Because they are for content size, which was truncated by border padding in block direction when `box-sizing: border-box`, thus will give incorrect result.

Reply:
I think we precisely want that truncated value.